presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKerberosKmsHdfsImpersonation.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.product.launcher.env.environment;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.tests.product.launcher.docker.DockerFiles;
+import io.prestosql.tests.product.launcher.env.Environment;
+import io.prestosql.tests.product.launcher.env.EnvironmentOptions;
+import io.prestosql.tests.product.launcher.env.common.AbstractEnvironmentProvider;
+import io.prestosql.tests.product.launcher.env.common.Hadoop;
+import io.prestosql.tests.product.launcher.env.common.Kerberos;
+import io.prestosql.tests.product.launcher.env.common.Standard;
+
+import javax.inject.Inject;
+
+import static io.prestosql.tests.product.launcher.env.common.Hadoop.CONTAINER_PRESTO_HIVE_PROPERTIES;
+import static io.prestosql.tests.product.launcher.env.common.Standard.CONTAINER_TEMPTO_PROFILE_CONFIG;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.BindMode.READ_ONLY;
+
+public final class SinglenodeKerberosKmsHdfsImpersonation
+        extends AbstractEnvironmentProvider
+{
+    private final DockerFiles dockerFiles;
+
+    private final String imagesVersion;
+
+    @Inject
+    public SinglenodeKerberosKmsHdfsImpersonation(DockerFiles dockerFiles, Standard standard, Hadoop hadoop, Kerberos kerberos, EnvironmentOptions environmentOptions)
+    {
+        super(ImmutableList.of(standard, hadoop, kerberos));
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        imagesVersion = requireNonNull(environmentOptions.imagesVersion, "environmentOptions.imagesVersion is null");
+    }
+
+    @Override
+    @SuppressWarnings("resource")
+    protected void extendEnvironment(Environment.Builder builder)
+    {
+        super.extendEnvironment(builder);
+
+        // TODO (https://github.com/prestosql/presto/issues/1652) create images with HDP and KMS
+        String dockerImageName = "prestodev/cdh5.15-hive-kerberized-kms:" + imagesVersion;
+
+        builder.configureContainer("hadoop-master", container -> {
+            container.setDockerImageName(dockerImageName);
+            container
```

Comment:
I find this wrapping strange. We don't normally do that with builders when the reference is a simple variable.

Reply:
I use it in other places too
https://github.com/prestosql/presto/blob/d047f22f3801eaf60042921e089c5758898b6d9d/presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/MultinodeTls.java#L68-L70

also, the advantage of this is that code is copy-pastable between `configureContainer` and methods creating new container.
this is useful when eg setting some networking properties for both: new & existing containers

Reply:
i have no hard feelings anyway. i am open, we can change it, if we want to